keras_cv/models/regnet.py

```diff
@@ -0,0 +1,1524 @@
+):
+    """Instantiates RegNet architecture given specific configuration.
+
+    Args:
+      depths: An iterable containing depths for each individual stages.
+      widths: An iterable containing output channel width of each individual
+        stages
+      group_width: Number of channels to be used in each group. See grouped
+        convolutions for more information.
+      block_type: Must be one of `{"X", "Y", "Z"}`. For more details see the
+        papers "Designing network design spaces" and "Fast and Accurate Model
+        Scaling"
+      default_size: Default input image size.
+      model_name: An optional name for the model.
+      include_rescaling: boolean denoting whther to include preprocessing in
+        the model
+      include_top: Boolean denoting whether to include classification head to
+        the model.
+      weights: one of `None` (random initialization) or the path to the weights file to be loaded.
+      input_shape: optional shape tuple, only to be specified if `include_top`
+        is False. It should have exactly 3 inputs channels.
+      pooling: optional pooling mode for feature extraction when `include_top`
+        is `False`. - `None` means that the output of the model will be the 4D
+        tensor output of the last convolutional layer. - `avg` means that global
+        average pooling will be applied to the output of the last convolutional
+        layer, and thus the output of the model will be a 2D tensor. - `max`
+        means that global max pooling will be applied.
+      num_classes: optional number of classes to classify images into, only to be
+        specified if `include_top` is True, and if no `weights` argument is
+        specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        Defaults to `"softmax"`.
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if not (weights is None or tf.io.gfile.exists(weights)):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` (random initialization) "
+            "or the path to the weights file to be loaded."
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}"
+        )
+
+    if include_top and pooling:
+        raise ValueError(
+            f"`pooling` must be `None` when `include_top=True`."
+            f"Received pooling={pooling} and include_top={include_top}. "
+        )
+
+    img_input = layers.Input(shape=input_shape)
+
+    x = img_input
+    if include_rescaling:
+        x = layers.Rescaling(scale=1.0 / 255.0)(x)
+    x = Stem(name=model_name)(x)
+
+    in_channels = 32  # Output from Stem
+
+    for num_stage in range(4):
```

Comment:
I assume this is a carryover from the Keras.applications impl, but this name is a bit hard to follow. Maybe something like `stage_number` would be easier to understand?

Reply:
Or `stage_index`.